[Ong Xinyi] iP

ongxinyii opened the pull request:
### **Pookie**

Pookie 🎀 is your personal assistant who is _cute_, _loving_ and _helpful_, fit for a princess like yourself. 
> If I am a princess in rags and tatters, I can be a princess inside. Frances Hodgson Burnett, [A Little Princess](https://www.goodreads.com/work/quotes/1313599)

They are:
- text-based
- simple and functional

You can download Pookie from [here](https://github.com/ongxinyii/ip/releases).

**What's New**:
1. Enhanced Task Management features
2. Searching Tasks by Keyword
3. Enhanced UI (see Pookie's unique emotions!)

With `enum` Included:
```java
package pookie.task;

public enum TaskType {
    TODO,
    DEADLINE,
    EVENT
}
``` 

Checklist (future updates):
- [x] Added find feature
- [x] Wrote unit tests for edge cases
- [ ] Update documentation to reflect new feature
- [ ] Add GUI

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ongxinyii committed:
Level-0

ongxinyii committed:
Level-1

ongxinyii committed:
Level-2

ongxinyii committed:
Level-3

ongxinyii committed:
Level-4

ongxinyii committed:
Level-5

ongxinyii committed:
Added deletion and handling of exceptions correctly

ongxinyii committed:
Level-6

ongxinyii committed:
Added local storage and deletion (with pookie.txt file with list)

ongxinyii committed:
Merge branch 'branch-Level-7'

ongxinyii committed:
Added time and dates in YYYY-MM-DD format

ongxinyii committed:
Merge branch 'branch-Level-8'

ongxinyii committed:
Added Ui, Storage, Parser and TaskList, more OOP

ongxinyii committed:
Divided classes into packages

ongxinyii committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ongxinyii committed:
JUnit tests added to all classess

ongxinyii committed:
Setting up JAR

ongxinyii committed:
Added JavaDoc to classes (public methods) and non trivial private methods

ongxinyii committed:
Add standardised coding standard

ongxinyii committed:
Add find feature for users to find tasks by a keyword

ongxinyii committed:
Merge branch 'branch-A-JavaDoc'

ongxinyii committed:
Merge branch 'branch-A-CodingStandard'

ongxinyii committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/pookie/list/TaskList.java
#	src/main/java/pookie/task/Task.java

ongxinyii committed:
Correct usuccessful merging due to merge conflicts

ongxinyii committed:
Corrected usuccessful merging due to merge conflicts

ongxinyii committed:
Merge branch 'master' of https://github.com/ongxinyii/ip

lilyium commented on src/main/java/pookie/command/Parser.java:
```diff
@@ -0,0 +1,198 @@
+import pookie.exception.PookieException;
+import pookie.list.TaskList;
+import pookie.storage.Storage;
+import pookie.task.*;
```
It might be better to list imported classes explicitly.

lilyium commented on src/main/java/pookie/command/Parser.java:
```diff
@@ -0,0 +1,198 @@
+            }
+
+            switch (type) {
+                case TODO:
+                    handleTodo(tasks, parts.length > 1 ? parts[1].trim() : "", ui, storage);
+                    break;
+                case DEADLINE:
+                    handleDeadline(tasks, parts.length > 1 ? parts[1].trim() : "", ui, storage);
+                    break;
+                case EVENT:
+                    handleEvent(tasks, parts.length > 1 ? parts[1].trim() : "", ui, storage);
+                    break;
+                default:
+                    throw new PookieException("Sowwieeee (╥﹏╥) Pookie doesn't understand...");
```
Case clauses should not be indented.

lilyium commented on src/main/java/pookie/command/Parser.java:
```diff
@@ -0,0 +1,198 @@
+
+        try {
+            switch (type) {
+                case "T": // ToDo task
+                    ToDo todo = new ToDo(description);
+                    if (isDone) todo.markDone();
+                    return todo;
+
+                case "D": // Deadline task
+                    if (parts.length < 4) throw new PookieException("Error: Missing deadline date.");
+                    LocalDateTime deadlineDate = LocalDateTime.parse(parts[3], DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
+                    Deadline deadline = new Deadline(description, deadlineDate);
+                    if (isDone) deadline.markDone();
+                    return deadline;
+
+                case "E": // Event task
+                    if (parts.length < 5) throw new PookieException("Error: Missing event start and end time.");
+                    LocalDateTime startDate = LocalDateTime.parse(parts[3], DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
+                    LocalDateTime endDate = LocalDateTime.parse(parts[4], DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
+                    Event event = new Event(description, startDate, endDate);
+                    if (isDone) event.markDone();
+                    return event;
+
+                default:
+                    throw new PookieException("Error: Unknown task type.");
```
Case clauses should not be indented.

lilyium commented on src/main/java/pookie/list/TaskList.java:
```diff
@@ -0,0 +1,133 @@
+package pookie.list;
+
+import pookie.task.*;
```
It might be better to list imported classes explicitly.

lilyium commented on src/main/java/pookie/storage/Storage.java:
```diff
@@ -0,0 +1,71 @@
+package pookie.storage;
+
+import pookie.task.*;
+import pookie.exception.PookieException;
+import pookie.command.Parser;
+
+import java.io.*;
```
It might be better to list imported classes explicitly.

lilyium commented on src/main/java/pookie/task/Event.java:
```diff
@@ -0,0 +1,87 @@
+            this.start = LocalDateTime.parse(start, INPUT_FORMAT);
+            this.end = LocalDateTime.parse(end, INPUT_FORMAT);
+        } catch (DateTimeParseException e) {
+            throw new PookieException("Princess, please enter the event times in the correct format: yyyy-MM-dd HHmm (e.g., 2019-12-02 1800).");
```
You might want to consider wrapping this line.

lilyium reviewed (commented):
LGTM :+1:! You might want to try to have the import statements in a consistent order throughout the files.

boinkkitty commented on src/main/java/pookie/command/Parser.java:
```diff
@@ -0,0 +1,198 @@
+    /**
+     * Handles creation of a ToDo task.
+     */
+    private static void handleTodo(TaskList tasks, String description, Ui ui, Storage storage) throws PookieException {
```
Could consider renaming to handleToDo to maintain consistency with ToDo class name

boinkkitty commented on src/main/java/pookie/command/Parser.java:
```diff
@@ -0,0 +1,198 @@
+        if (parts[0].trim().isEmpty()) {
+            throw new PookieException.EmptyDescriptionException("Princess, the description of a deadline cannot be empty.");
+        }
+        if (parts.length < 2 || parts[1].trim().isEmpty()) {
```
I think coding convention requires it to be in an else if and else block

boinkkitty commented on src/main/java/pookie/command/Parser.java:
```diff
@@ -0,0 +1,198 @@
+     */
+    private static void handleEvent(TaskList tasks, String details, Ui ui, Storage storage) throws PookieException {
+        String[] parts = details.split(" /from | /to ");
+        if (parts[0].trim().isEmpty()) {
```
I think coding convention requires it to be in an else if and else block

boinkkitty commented on src/main/java/pookie/command/Parser.java:
```diff
@@ -0,0 +1,198 @@
+    /**
+     * Lists tasks occurring on a specific date.
+     */
+    private static void handleListByDate(TaskList tasks, String dateStr, Ui ui) {
```
Cool method to search by date

boinkkitty commented on src/main/java/pookie/exception/PookieException.java:
```diff
@@ -0,0 +1,28 @@
+    public PookieException(String message) {
+        super(message);
+    }
+
```
I think Javadoc comments should be in /** format

boinkkitty commented on src/main/java/pookie/task/Task.java:
```diff
@@ -0,0 +1,57 @@
+     *
+     * @return A formatted string representation of the task.
+     */
+    public abstract String toFileFormat();
```
Nice use of abstract class and method

boinkkitty commented on src/main/java/pookie/task/TaskType.java:
```diff
@@ -0,0 +1,7 @@
+package pookie.task;
+
+public enum TaskType {
```
Nice use of enum

boinkkitty commented on src/main/java/pookie/ui/Ui.java:
```diff
@@ -0,0 +1,84 @@
+     * Prints a horizontal border line for message separation.
+     */
+    private void printBorder() {
+        System.out.println("____________________________________________________________");
```
Could consider initializing as a constant like 'DIVIDER' if you prefer

boinkkitty reviewed (commented):
LGTM. Just minor things like packaging and if else blocks.

meglwhy commented:
LGTM! just minor nits but coding style is good

ongxinyii committed:
Fix styling errors

ongxinyii committed:
Add GUI to Pookie Chatbot

ongxinyii committed:
Fix format errors

ongxinyii committed:
Merge branch 'branch-Level-10'

ongxinyii committed:
Add assertion and bug fixes

ongxinyii committed:
Merge branch 'branch-A-Assertions'

ongxinyii committed:
Fix code quality issues

ongxinyii committed:
Fix incorrect parsing of fixed_duration tasks

FixedDurationTask commands were incorrectly being handled as events,
causing an error message requiring "/from" and "/to" fields. This was
due to executeCommand() calling handleEvent() instead of
handleFixedDurationTask().

This issue prevented users from adding fixed duration tasks and
resulted in misleading error messages.

Let's fix this by:
* Ensuring executeCommand() correctly calls handleFixedDurationTask()
* Updating parseTaskFromLine() to properly reconstruct FixedDurationTask
* Improving error handling when /duration is missing or incorrect

These changes ensure that fixed duration tasks are parsed and stored
correctly, allowing users to create them as expected.

ongxinyii committed:
Merge branch 'branch-A-CodeQuality'

ongxinyii committed:
Add fixed duration tasks

ongxinyii committed:
Merge pull request #2 from ongxinyii/branch-B-FixedDurationTasks

Merge b fixed duration tasks

ongxinyii committed:
Fix errors with list on command

ongxinyii committed:
Merge branch 'master' of https://github.com/ongxinyii/ip

ongxinyii committed:
Merge branch 'branch-B-FixedDurationTasks'

ongxinyii committed:
Update README.md

ongxinyii committed:
Add files via upload

ongxinyii committed:
Update README.md

ongxinyii committed:
Update README.md

ongxinyii committed:
Fix error with bye command and update README.md

ongxinyii committed:
Merge branch 'master' of https://github.com/ongxinyii/ip
